tpke/src/ciphertext.rs

```diff
@@ -25,13 +25,43 @@ impl<E: PairingEngine> Ciphertext<E> {
             (g_inv.clone(), E::G2Prepared::from(self.auth_tag)),
         ]) == E::Fqk::one()
     }
+
     fn construct_tag_hash(&self) -> E::G2Affine {
         let mut hash_input = Vec::<u8>::new();
         self.commitment.write(&mut hash_input).unwrap();
         hash_input.extend_from_slice(&self.ciphertext);
 
         hash_to_g2(&hash_input)
     }
+
+    pub fn to_bytes(&self) -> Vec<u8> {
+        let mut bytes = Vec::new();
+        self.nonce.write(&mut bytes).unwrap();
+        bytes.extend_from_slice(&self.ciphertext);
+        self.auth_tag.write(&mut bytes).unwrap();
+        bytes
+    }
+
+    pub fn from_bytes(bytes: &[u8]) -> Self {
+        const NONCE_LEN: usize = 97;
+        let mut nonce_bytes = [0u8; NONCE_LEN];
+        nonce_bytes.copy_from_slice(&bytes[..NONCE_LEN]);
+        let nonce = E::G1Affine::read(&nonce_bytes[..]).unwrap();
+
+        const CIPHERTEXT_LEN: usize = 33;
```

Comment:
Why is it 33 bytes? The ciphertext length shouldn't be constant as it varies with the message to be encrypted. Related to this, I'm thinking that the serialization order should be `commitment, auth_tag, ciphertext` since the first 2 components have fixed size.

Reply:
But the ciphertext size depends on the message, not on arkworks stuff, right?

Reply:
Sorry, I misunderstood your question - I thought you were referring to `struct Ciphertext<E: PairingEngine>`.

Yes, the ciphertext size depends on the message. `const CIPHERTEXT_LEN: usize = 33;` is not actually used anywhere! It's a left-over artifact of some refactoring that I missed. For some reason, it wasn't picked up by the linter. I'm going to remove this line.